### PR TITLE
TEZ-4478: Refactor dependency versions into properties in pom.

### PR DESCRIPTION
**Pull request by ayushtkn**

### CODE REVIEW

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  23m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  1s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m  3s |  master passed  |
| +1 :green_heart: |  compile  |   7m 47s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   7m 30s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javadoc  |   7m 27s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   6m 34s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 51s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   7m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 33s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javac  |   7m 33s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 13s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   7m  7s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   6m 43s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 31s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 54s |  tez-runtime-internals in the patch passed.  |
| +1 :green_heart: |  unit  |   6m  7s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 39s |  tez-mapreduce in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 27s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 19s |  tez-ext-service-tests in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 43s |  tez-protobuf-history-plugin in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  7s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 26s |  docs in the patch passed.  |
| +1 :green_heart: |  unit  |  75m  0s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   5m  2s |  The patch does not generate ASF License warnings.  |
|  |   | 211m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/291 |
| JIRA Issue | TEZ-4478 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 5806ec24307a 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 86b11997b |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/1/testReport/ |
| Max. process+thread count | 2100 (vs. ulimit of 5500) |
| modules | C: tez-api tez-runtime-internals tez-runtime-library tez-mapreduce tez-dag tez-ext-service-tests tez-plugins/tez-protobuf-history-plugin tez-plugins/tez-aux-services docs . U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/1/console |
| versions | git=2.34.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  23m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 41s |  master passed  |
| +1 :green_heart: |  compile  |   7m 49s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   7m  4s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javadoc  |   8m  6s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   6m 46s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   8m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 47s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   7m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 22s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javac  |   7m 22s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 13s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   7m 18s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   6m 45s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 26s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 54s |  tez-runtime-internals in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 59s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 41s |  tez-mapreduce in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 27s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 22s |  tez-ext-service-tests in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 42s |  tez-protobuf-history-plugin in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  8s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  docs in the patch passed.  |
| +1 :green_heart: |  unit  |  74m 26s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   4m 55s |  The patch does not generate ASF License warnings.  |
|  |   | 216m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/291 |
| JIRA Issue | TEZ-4478 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux ac19ad52943d 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 86b11997b |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/2/testReport/ |
| Max. process+thread count | 2100 (vs. ulimit of 5500) |
| modules | C: tez-api tez-runtime-internals tez-runtime-library tez-mapreduce tez-dag tez-ext-service-tests tez-plugins/tez-protobuf-history-plugin tez-plugins/tez-aux-services docs . U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/2/console |
| versions | git=2.34.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -56,26 +56,50 @@
     <distMgmtStagingName>Apache Release Distribution Repository</distMgmtStagingName>
     <distMgmtStagingUrl>https://repository.apache.org/service/local/staging/deploy/maven2</distMgmtStagingUrl>
 
+<!--    dependency versions in alphabetical order-->
```
Comment (abstractdog) [inline pom.xml:59]: nit: indent comment with properties

**Review by abstractdog — APPROVED** [review]
minor nit, other than that +1

**Review by abstractdog — APPROVED** [review]

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 19s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 37s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 50s |  master passed  |
| +1 :green_heart: |  compile  |   7m 49s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   7m 31s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javadoc  |   5m 53s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   5m 16s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 52s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   7m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 24s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javac  |   7m 24s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 13s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   5m 25s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   5m 27s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 31s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 54s |  tez-runtime-internals in the patch passed.  |
| +1 :green_heart: |  unit  |   6m  4s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 39s |  tez-mapreduce in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 26s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 31s |  tez-ext-service-tests in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 40s |  tez-protobuf-history-plugin in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  6s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 25s |  docs in the patch passed.  |
| +1 :green_heart: |  unit  |  74m 58s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   5m  2s |  The patch does not generate ASF License warnings.  |
|  |   | 187m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/291 |
| JIRA Issue | TEZ-4478 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux c08df1688944 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / a1a8f2050 |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/3/testReport/ |
| Max. process+thread count | 2100 (vs. ulimit of 5500) |
| modules | C: tez-api tez-runtime-internals tez-runtime-library tez-mapreduce tez-dag tez-ext-service-tests tez-plugins/tez-protobuf-history-plugin tez-plugins/tez-aux-services docs . U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-291/3/console |
| versions | git=2.34.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**